aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -106,82 +95,16 @@ public String endpoint() {
     return properties().getOrDefault(S3_SIGNER_ENDPOINT, S3_SIGNER_DEFAULT_ENDPOINT);
   }
 
-  /** A credential to exchange for a token in the OAuth2 client credentials flow. */
-  @Nullable
-  @Value.Lazy
-  public String credential() {
-    return properties().get(OAuth2Properties.CREDENTIAL);
-  }
-
-  /** Token endpoint URI to fetch token from if the Rest Catalog is not the authorization server. */
-  @Value.Lazy
-  public String oauth2ServerUri() {
-    return properties().getOrDefault(OAuth2Properties.OAUTH2_SERVER_URI, ResourcePaths.tokens());
-  }
-
-  @Value.Lazy
-  public Map<String, String> optionalOAuthParams() {
-    return OAuth2Util.buildOptionalParam(properties());
-  }
-
-  /** A Bearer token supplier which will be used for interaction with the server. */
-  @Value.Default
-  public Supplier<String> token() {
```

Comment:
removing this will actually break existing applications that were providing the token through a supplier